### PR TITLE
AllowExplicitVersion if referencing a WebApi project

### PR DESCRIPTION
**Pull request by jzabroski**
@mairaw  See my comments on https://github.com/dotnet/sdk/issues/2879#issuecomment-476811104

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@balachir can you recommend someone to review this PR?

**Comment by balachir** [comment]
From https://github.com/aspnet/Docs/issues/9490, I know the proposed change applies to 2.2 to 3.0 migration. But this PR is about 2.1 to 2.2 migration, I'm not sure if it applies here as well. I think @natemcmaster @JunTaoLuo could be the right people to review.

**Comment by jzabroski** [comment]
Nate is on paternity leave. #9490 is interesting but Nate didn't include
the compiler errors so most end users like me couldn't Google search for
it. I'm off today but can try tomorrow to see if FrameworkReference works.
If it does I'll update the PR

**Comment by JunTaoLuo** [comment]
Seems like from https://github.com/dotnet/sdk/issues/2602, @dsplaisted might be able to help here.

File: aspnetcore/migration/21-to-22.md [hunk 1]
```diff
@@ -68,6 +68,8 @@ The metapackage reference should resemble the following `<PackageReference />` n
 </ItemGroup>
 ```
 
+When targeting .NET Core and transitively referencing a .NET Core project in another .NET Core project (such as a unit test project), implicit references will map to 2.1 assemblies, resulting in build errors.  In this case, add `<PackageReference Include="Microsoft.AspNetCore.App" />` as a package reference in your project (e.g., the unit test project referencing ASP.NET Core metapackage).
```
Comment (dsplaisted) [inline aspnetcore/migration/21-to-22.md]: I still don't think implicit references will map to 2.1 assemblies.  I would expect that you just wouldn't get compile-time references to ASP.NET at all.

What is the behavior you're seeing that makes you think 2.1 assemblies are involved?

Comment (jzabroski) [reply aspnetcore/migration/21-to-22.md]: Pivoted the Visual Studio Errors window copied values into a format more readable on GitHub (wish this is how Visual Studio copying of this grid worked by default!)

```
Severity: Error		
Code: CS1705
Description: Assembly '.WebApi' with identity 'WebApi, Version=1.5.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.ViewFeatures, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.ViewFeatures' with identity 'Microsoft.AspNetCore.Mvc.ViewFeatures, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
Project: WebApi.Tests
File: D:\source\John.Zabroski\Source\WebApi.Tests\CSC
Line:	 1
Suppression State: Active
```

This happens even if I clean bin and obj directories.  It also happens even if I add the implicit package reference in Tests, build successfully, remove it without cleaning bin and obj directories, and build again.

**Comment by Rick-Anderson** [comment]
@jzabroski  what can I do to help get this approved?

**Comment by jzabroski** [comment]
I think I need to Uninstall Incredibuild or whatever is confusing my repro of the issue. Daniel is advising I reference the full package which I'm not in love with for many reasons. The obvious being it wastes disk space especially if you run build servers are nano containers.

What are your thoughts?

**Comment by dsplaisted** [comment]
The full package is part of the .NET Core SDK, so you referencing it should not take up additional disk space past the SDK install.

**Comment by jzabroski** [comment]
Excellent point that is obvious after you mention it.

@Rick-Anderson  I will close this pull request.  I will create a new one that instead does the workaround Daniel suggests.

TYVM for your help and patience with me, @dsplaisted  and @Rick-Anderson

**Pull request closed without merging**